Automattic/jetpack
# Update/conditionally hide connection banner

## Proposed changes:

* Only show site connection banner if the site previously had working products that require the site connection but not the user connection
* Only show the user connection banner if the site previously had working products that require the user connection

### Other information:


## Jetpack product discussion

pbNhbs-akp-p2

## Does this pull request change what data or activity we track or use?

Yes, we are adding some tracks attributes to the `jetpack_myjetpack_global_notice_view` event such as the type of connection issue

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
(In this case I would suggest Jetpack Beta so you can more easily start with a fresh site)
2. Go to My Jetpack and dismiss the welcome banner. Refresh the page. You should see a connection banner with a "info" display rather than an error
![image](https://github.com/Automattic/jetpack/assets/65001528/348da8f1-899f-4d03-ad4d-91469c868c83)
3. Connect your site (not the user)
4. Go back to My Jetpack and you should still see a user connection banner with an info banner rather than an error banner
![image](https://github.com/Automattic/jetpack/assets/65001528/7a456659-b6ae-4428-928b-3b790b71060d)
6. Disconnect your site from Jetpack Debugger
7. Go back to My Jetpack and you should now see the site connection banner with error styles
![image](https://github.com/Automattic/jetpack/assets/65001528/f39c08f7-5b35-4491-9867-04cac58b6acf)
8. Connect your site from the banner and refresh the page, you should see the user connection info banner again
9. Now connect your user account
10. Disconnect your site again and go back to My Jetpack. You should now see the *user* error connection banner instead of the site connection banner since you previously had a working plugin that requires a user connection (Jetpack AI)
![image](https://github.com/Automattic/jetpack/assets/65001528/bad2611e-0134-42cf-894d-90245e9afeda)
